src/replicache.ts

```diff
@@ -291,8 +294,13 @@ export class Replicache<MD extends MutatorDefs = {}>
     // wait for it to finish closing.
     await closingInstances.get(this._name);
 
+    const store = new WrapStore(
```

Comment:
Do we want to rely on the IDB locking mechanism? The current way we end up ensuring things are in an RWLock we control.

WrapStore (renamed to ReleasableStore is not part of the public API)

Reply:
> The current way we end up ensuring things are in an RWLock we control.

I remember talking about this and @phritz arguing for it, but now that I look at it again, I really don't like it - particularly if we are exposing it to users.

A key property of Replicache is correctness and I don't like the squishiness of "the semantics within one tab are a RWLock but across tabs the semantics are ... whatever IDB does".

I think @phritz argument at the time was that the RW Lock was clean and easy to understand, whereas IDB might permit looser models depending on how one reads the spec. But looking at latest version of the spec now, the desired behavior seems very clearly to be a RW lock:

<img width="898" alt="Screen Shot 2021-08-03 at 12 26 36 AM" src="https://user-images.githubusercontent.com/80388/128000723-0f715081-5423-481d-b1c7-9d9c98b4c17c.png">

However, no browsers today implement IDB v3. So the question is whether this change to the spec is meant to be a semantic change or is just a clarification. I am betting it is the latter and that browsers actually implement per-object store RW locks today, since that's the sensible thing.

I think we should confirm this then remove the in-memory RW lock from the IDB path.

Reply:
I'm also a bit concerned about how to get the current promise based locking behavior. Maybe it is not needed and maybe we do not depend on it but I bet my ass that our tests depends on it.